lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RMS.pm

```diff
@@ -0,0 +1,89 @@
+package GLPI::Agent::Task::Inventory::Generic::Remote_Mgmt::RMS;
+
+use strict;
+use warnings;
+
+use parent 'GLPI::Agent::Task::Inventory::Module';
+
+use English qw(-no_match_vars);
+
+use GLPI::Agent::Tools;
+
+sub isEnabled {
+    my (%params) = @_;
+
+    if (OSNAME eq 'MSWin32') {
+
+        GLPI::Agent::Tools::Win32->use();
+
+        my $key = getRegistryKey(
+            path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Usoris/Remote Utilities Host/Host/Parameters',
+            # Important for remote inventory optimization
+            required    => [ qw/InternetId/ ],
+            maxdepth    => 1,
+            logger => $params{logger}
+        );
+        # if (!$key && is64bit()) {
+        #     $key = getRegistryKey(
+        #         path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Usoris/Remote Utilities Host/Host/Parameters',
+        #         # Important for remote inventory optimization
+        #         required    => [ qw/InternetId/ ],
+        #         maxdepth    => 1,
+        #         logger => $params{logger}
+        #     );
+        # }
+
+        return 1;
+    }
+
+    return canRun('rms');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $InternetID = OSNAME eq 'MSWin32' ? _getID_MSWin32(logger  => $logger) : _getID(logger  => $logger);
+    );
+    if (defined($InternetID)) {
+        $logger->debug('Found InternetID : ' . $InternetID) if ($logger);
+
+        $inventory->addEntry(
+            section => 'REMOTE_MGMT',
+            entry   => {
+                ID   => $InternetID,
+                TYPE => 'rms'
+            }
+        );
+    } else {
+        $logger->debug('InternetID not found') if ($logger);
+    }
+}
+
+# sub _getID {
+#     my (%params) = @_;
+#     my $osname = delete $params{osname} // '';
+#     return _getID_MSWin32() if $osname eq "MSWin32";
+# }
+
+sub _getID_MSWin32 {
+
+    GLPI::Agent::Tools::Win32->use();
+    GLPI::Agent::XML->use();
+
+    my $internetid = getRegistryValue(
+        path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Usoris/Remote Utilities Host/Host/Parameters/InternetId',
+    );
+
+    $internetid = hex2dec($clientid);
+
+    my $tree = GLPI::Agent::XML->new(string => $internetid)->dump_as_hash();
+
+    return unless defined($tree) && defined($tree->{nodes_path_to_internet_id});
+
+    return $tree->{nodes_path_to_internet_id}->{internet_id};
```

Comment:
You still need here to find the `internet_id` node the the `$tree` structure.

Reply:
Help me do this, I'm not good at Perl. Here is the xml text from the registry:
﻿<?xml version="1.0" encoding="UTF-8"?>
<rms_internet_id_settings version="70170"><internet_id>891-759-358-441</internet_id><use_inet_connection>true</use_inet_connection><inet_server></inet_server><use_custom_inet_server>false</use_custom_inet_server><inet_id_port>5655</inet_id_port><use_inet_id_ipv6>false</use_inet_id_ipv6><inet_id_use_pin>false</inet_id_use_pin><inet_id_pin></inet_id_pin></rms_internet_id_settings>

Reply:
Okay, here is the code I think you should try:
```suggestion
    return unless defined($tree) && defined($tree->{rms_internet_id_settings});

    return $tree->{rms_internet_id_settings}->{internet_id};
```